sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,207 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[=[
+Provides functions to convert full Heka message contents to line protocol for
+InfluxDB HTTP write API (new in InfluxDB v0.9.0) or Graphite/Carbon.
+Optionally includes all standard message fields as tags or fields and
+iterates through all of the dynamic fields
+to add as points (series), skipping any fields explicitly omitted using the
+`skip_fields` config option.  It can also map any Heka message fields as tags
+in the request sent to the InfluxDB write API, using the `tag_fields` config
+option.  All dynamic fields in the Heka message are converted to separate points
+separated by newlines that are submitted to InfluxDB.
+
+API
+^^^
+
+**carbon_line_msg()**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of Carbon line protocol messages that are
+    derived from the dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message.
+    Configuration is implemented in the encoders that utilize this module.
+    This function takes no arguments.
+
+**influxdb_line_msg()**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of InfluxDB line protocol messages that are
+    derived from the base or dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message and
+    included as tags if desired.  Configuration is implemented in the encoders
+    that utilize this module.  This function takes no arguments.
+
+--]=]
```

Comment:
Seems like this should probably document the attribute portion of the API, since AFAICT this module won't really work correctly unless you set the attribute values that it expects. Also it would be nice to explicitly define the settable variables in the code below, with a comment explaining that the values are expected to be mutated by calling code prior to any use of the API functions.